x/inflation/types/v1beta2/inflation_calculator.go

```diff
@@ -0,0 +1,68 @@
+package v1beta2
+
+import (
+	"math"
+	"strconv"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+var (
+	// these are set during NewApp and used in InflationCalculator
+
+	GenesisTime            time.Time
+	InflationParamSubspace paramstypes.Subspace
+)
+
+func InflationCalculator(ctx sdk.Context, minter minttypes.Minter, params minttypes.Params, bondedRatio sdk.Dec) sdk.Dec {
+	var inflationParams Params
+	InflationParamSubspace.GetParamSet(ctx, &inflationParams)
+	tHalf := float64(inflationParams.InflationDecayFactor)
+	initialInflation, err := strconv.ParseFloat(inflationParams.InitialInflation, 32)
+	if err != nil {
+		panic(err)
+	}
+
+	// Number of hours in an year = 8766 (averaging the leap year hours)
+	// Number of minutes in an hour = 60
+	// Number of seconds in a minute = 60
+	// => 60 * 60 * 8766 = Number of seconds per year
+	t := ctx.BlockTime().Sub(GenesisTime).Seconds() / (60 * 60 * 8766) // years passed
+	i := initialInflation * math.Pow(2, -t/tHalf)
+	idealInflation := sdk.NewDec(int64(i))
```

Comment:
This converts an int to a decimal number, which implies some loss of precision. Whatever the intended operation is here should be documented in a comment.

Reply:
I don't think that there is any loss of precision from int to decimal.

I do think that we should use a consistent type from the outset of this whole thing - rational and decimal for arbitrary precision, then truncate as necessary at the end (or continuously via some large fixed precision).

Reply:
My statement was accurate. It converts a float, to an int, to a decimal. We should try and document the conversion from float to decimal or avoid the whole thing.

Reply:
Done